### PR TITLE
tracing: support SDS for extracting backend token on Apache SkyWalking

### PR DESCRIPTION
**Pull request by Shikugawa**
Signed-off-by: Shikugawa <rei@tetrate.io>

Commit Message: tracing: support SDS for extracting backend token on Apache SkyWalking
Additional Description: This PR allows to extract backend token via SDS.
Risk Level: Low
Testing: Unit
Docs Changes: Required
Release Notes: Required
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]  https://github.com/istio/istio/pull/35972#issuecomment-964749046
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by lizan** [comment]
@Shikugawa ping me when CI passes.

**Comment by Shikugawa** [comment]
@lizan Could you take a look?

File: test/per_file_coverage.sh
Comment (lizan) [inline test/per_file_coverage.sh:28]: What's the cause of this being lower? Can you add more tests?

Comment (Shikugawa) [reply test/per_file_coverage.sh:28]: This fix came from this change. https://github.com/envoyproxy/envoy/pull/19030/files#diff-b34e5ba9bf6eb09787a43f6a1f8ccead0e9e7f412eaa12c2b7d06f7d4d150ba2R36
We are using `TracerFactoryContext::transportSocketFactoryContext()` only [here](https://github.com/envoyproxy/envoy/pull/19030/files#diff-dfccda64c35905772b119e9525441cb73ff37a85353e8d0dac4b0dfb960e848fR76-R77). But, we are using `TracerFactoryContextMock` in the tracing test. So we can't cover those lines in the current implementation.

**Comment by phlax** [comment]
@lizan @Shikugawa is this ready for final review ?

**Comment by Shikugawa** [comment]
@lizan @phlax Yes. It has been waiting for review...

**Review by lizan — APPROVED** [review]
LGTM, need a non-Tetrands maintainer review.

File: test/per_file_coverage.sh
Comment (htuch) [reply test/per_file_coverage.sh:28]: Can you add an integration test to cover this?

**Review by htuch — COMMENTED** [review]
I'm unable to view the coverage report. Implementation makes sense, but I'd like to verify that the test coverage exists for the relevant code.

**Comment by htuch** [comment]
@Shikugawa can you look into the lowered coverage and also do an empty/dummy push to trigger coverage rerun? Thanks.

**Pull request closed without merging**
